patches/0004-desktop-editor-intigration.patch

```diff
@@ -0,0 +1,209 @@
++    exists = await pathExists('/var/host/run'.toString().concat(editorPath))
++  } else {
++    exists = await pathExists(editorPath)
++  }
++  console.log(exists)
+   if (!exists) {
+     const label = __DARWIN__ ? 'Preferences' : 'Options'
+     throw new ExternalEditorError(
+@@ -30,12 +36,20 @@ export async function launchExternalEditor(
+   }
+ 
+   if (editor.usesShell) {
+-    spawn(`"${editorPath}"`, [`"${fullPath}"`], { ...opts, shell: true })
++    if (process.env.FLATPAK_HOST !== 'undefined') {
++      spawn('flatpak-spawn', ['--host', `"${editorPath}"`, `"${fullPath}"`], { ...opts, shell: true })
++    } else {
++      spawn(`"${editorPath}"`, [`"${fullPath}"`], { ...opts, shell: true })
++    }
+   } else if (__DARWIN__) {
+     // In macOS we can use `open`, which will open the right executable file
+     // for us, we only need the path to the editor .app folder.
+     spawn('open', ['-a', editorPath, fullPath], opts)
+   } else {
+-    spawn(editorPath, [fullPath], opts)
++    if (process.env.FLATPAK_HOST !== 'undefined') {
++      spawn('flatpak-spawn', ['--host', editorPath, fullPath], opts)
++    } else {
++      spawn(editorPath, [fullPath], opts)
++    }
+   }
+ }
+diff --git a/app/src/lib/editors/linux.ts b/app/src/lib/editors/linux.ts
+index 314344dde..7432182e1 100644
+--- a/app/src/lib/editors/linux.ts
++++ b/app/src/lib/editors/linux.ts
+@@ -34,7 +34,7 @@ const editors: ILinuxExternalEditor[] = [
+   },
+   {
+     name: 'VSCodium',
+-    paths: ['/usr/bin/codium','/var/lib/flatpak/app/com.vscodium.codium'],
++    paths: ['/usr/bin/codium', '/var/lib/flatpak/app/com.vscodium.codium', '/usr/share/vscodium-bin/bin/codium'],
+   },
+   {
+     name: 'Sublime Text',
+@@ -62,12 +62,26 @@ const editors: ILinuxExternalEditor[] = [
+ ]
+ 
+ async function getAvailablePath(paths: string[]): Promise<string | null> {
+-  for (const path of paths) {
+-    if (await pathExists(path)) {
+-      return path
++  if (process.env.FLATPAK_HOST !== 'undefined') {
++    for (const path of paths) {
++      console.log(path)
++      if (path.indexOf("/usr/") > -1) {
++        if (await pathExists("/var/run/host".toString().concat(path))) {
++          return path
++        }
++      } else {
++        if (await pathExists(path)) {
++          return path
++        }
++      }
++    }
++  } else {
++    for (const path of paths) {
++      if (await pathExists(path)) {
++        return path
++      }
+     }
+   }
+-
+   return null
+ }
+ 
+diff --git a/app/src/lib/shells/linux.ts b/app/src/lib/shells/linux.ts
+index 47a5c7f03..a73a9ffa8 100644
+--- a/app/src/lib/shells/linux.ts
++++ b/app/src/lib/shells/linux.ts
+@@ -27,7 +27,12 @@ export function parse(label: string): Shell {
+ }
+ 
+ async function getPathIfAvailable(path: string): Promise<string | null> {
+-  return (await pathExists(path)) ? path : null
++  if (process.env.FLATPAK_HOST !== 'undefined') {
++    var path_to_search: string = "/var/run/host".toString().concat(path)
```

Comment:
🤔 this should use the `path.join` method from NodeJS, which will handle whether or not `path` starts with a `/`:

```ts
import { join } from 'path'

...

const pathUnderHost = join("/var/run/host", path)
```

Reply:
Will change